frontend/src/pages/admin/user-list.js

```diff
@@ -9,229 +8,190 @@ import {
+                  setIsDeleteModalOpen(true);
+                }}
+                size="sm"
+              >
+                <TrashIcon />
+              </Button>
+            </div>
+          )
+        }
+      ]
     };
-  }
-
-  updateUrl () {
-    let params = [];
-    params.push('page=' + this.state.page);
-    params.push('pageSize=' + this.state.pageSize);
-    this.props.navigate('/admin/users?' + params.join('&'));
-  }
-
-  setPage = (_event, pageNumber) => {
-    this.setState({page: pageNumber}, () => {
-      this.updateUrl();
-      this.getUsers();
-    });
   };
 
-  setPageSize = (_event, perPage) => {
-    this.setState({pageSize: perPage}, () => {
-      this.updateUrl();
-      this.getUsers();
-    });
-  };
-
-  getUsers () {
-    // Show a spinner
-    this.setState({rows: [getSpinnerRow(4)], isEmpty: false, isError: false});
-    let params = {
-      pageSize: this.state.pageSize,
-      page: this.state.page
-    };
-    if (this.state.textFilter) {
-      params['filter'] = ['name%' + this.state.textFilter];
-    }
-    HttpClient.get([Settings.serverUrl, 'admin', 'user'], params)
+  useEffect(() => {
+    // fetch the users
+    HttpClient.get([Settings.serverUrl, 'admin', 'user'], {page: page, pageSize: pageSize})
       .then(response => HttpClient.handleResponse(response))
-      .then(data => this.setState({
-        rows: data.users.map((user) => userToRow(user, this.onDeleteClick)),
-        users: data.users,
-        page: data.pagination.page,
-        pageSize: data.pagination.pageSize,
-        totalItems: data.pagination.totalItems,
-        totalPages: data.pagination.totalPages,
-        isEmpty: data.pagination.totalItems === 0
-      }))
+      .then(data => {
+        if (data.users) {
+          setUsers(data.users);
+          setPage(data.pagination.page);
+          setPageSize(data.pagination.pageSize);
+          setTotalItems(data.pagination.totalItems);
+          setIsEmpty(false);
+        } else {
+          setIsEmpty(true);
+        }
+      })
       .catch((error) => {
         console.error('Error fetching users data:', error);
-        this.setState({rows: [], isEmpty: false, isError: true});
+        setUsers([]);
+        setIsError(true);
       });
-  }
+  }, [page, pageSize, isDeleting]); // isDeleteing so the users fetch after delete
 
-  onDeleteClick = (userId) => {
-    const selectedUser = this.state.users.find((user) => user.id === userId);
-    this.setState({selectedUser: selectedUser, isDeleteModalOpen: true});
-  };
+  useEffect(() => {
+    // filter fetched users
+    let newUsers = [...users];
+    if (textFilter && users) {
+      newUsers = users.filter((u) => String(u.name).toLowerCase().includes(textFilter.toLowerCase()));
+    }
 
-  onDeleteModalClose = () => {
-    this.setState({isDeleteModalOpen: false});
-  };
+    if (textFilter && newUsers.length === 0) {
+      newUsers = [{
+        name: `No results found for "${textFilter}"`
+      }];
+    }
```

Comment:
Same thing that I mentioned in projectList PR, when filter is applied and no results found - we need to show correct empty state